sdks/python/apache_beam/io/fileio.py

```diff
@@ -697,8 +680,7 @@ def close(self):
     if self._file is None:
       return
 
-    if self._writeable():
-      self._file.write(self._compressor.flush())
+    self.flush()
```

Comment:
But this means we're calling self._file.flush() when closing non-writable files which seems odd.